src/main/java/org/springframework/cloud/deployer/spi/mesos/marathon/MarathonAppDeployer.java

```diff
@@ -112,7 +111,7 @@ public String deploy(AppDeploymentRequest request) {
 
 		app.setCpus(cpus);
 		app.setMem(memory);
-		app.setInstances(Integer.getInteger(request.getDefinition().getProperties().get(COUNT_PROPERTY_KEY)));
+		app.setInstances(Integer.getInteger(request.getDeploymentProperties().get(COUNT_PROPERTY_KEY)));
```

Comment:
Actually, test for instance count should go in abstrat deployer TCK. There are also opportunities for more tests now that deploymentProperties contract has been clarified. Will open another issue